### ISSUE TITLE
[Bug]: isAuthenticated() not emitting after login in standalone setup

### ISSUE THREAD

**Issue by Spinnenzunge** [damienbod/angular-auth-oidc-client#1846]
### Version

16.0.1

### Please provide a link to a minimal reproduction of the bug

https://github.com/damienbod/angular-auth-oidc-client/tree/main/projects/sample-code-flow-standalone

### Please provide the exception or error you saw

```true
Methods like `isAuthenticated()`, `getUserData()`, `getPayloadFromIdToken()`, etc. seem to be broken in an Angular Standalone App Component setup. They do not emit latest value after login if they get called from AppComponent (or imported component, e.G. NavigationComponent)
```

### Steps to reproduce the behavior

```true
- Checkout https://github.com/damienbod/angular-auth-oidc-client/
- Start sample-code-flow-standalone
- Change from `isAuthenticated$` to `isAuthenticated()` in `NavigationComponent`
- Login
- Header does not change, because isAuthenticated() does not emit value after authentication
```

### A clear and concise description of what you expected to happen.

```true
Methods like `isAuthenticated()`, `getUserData()`, `getPayloadFromIdToken()`, etc. should always emit next values in stream in an Angular Standalone setup.
```

**Comment by Spinnenzunge** [comment]
Changed title and description as my initial issue was a bit confusing.

**Comment by dbl81** [comment]
same problem here

**Comment by FabianGosebrink** [comment]
When you call the methods again, do they emit the correct value?

**Comment by dbl81** [comment]
> When you call the methods again, do they emit the correct value?

Yes, but you have to "re-subscribe" in order to get the correct value, it doesnt stream the "new" value.
I use it in the ngOnInit of the app.component in order to show / not show user's initials in the upper right corner. If I reload the app (F5), isAuthenticated emit the correct value.

Subscribing to isAuthenticated() means to me:  
1. at first you are not authenticated -> initiallly it emits false
2. after login -> it emits true
3. logout -> it emits false

Did I misunderstand the use of isAuthenticated()?

I'm using:
* Angular 17.2.0
* angular-auth-oidc-client 17.0.0

**Comment by FabianGosebrink** [comment]
The intention of the method was: You call it, you get the value when you want it, once. We also have the other stream which should emit the value like a stream. So you can choose on your end what you want.

**Comment by dbl81** [comment]
Thank you for your support.

The problem with isAuthenticated$ comes up when you change route (it doesnt matter if it's protected or not by a AutoLoginPartialRoutesGuard) or you refresh the page, because the next value emitted is again false, even if I'm still authenticated (I can surf protected routes).

So in my case:  
1. isAuthenticated$ subscribed in app.component -> first value false (I mean the value of the property isAuthenticated of the AuthenticatedResult object)
2. hit a protected route by AutoLoginPartialRoutesGuard -> redirect to keycloak to login
3. after a successful login, redirect back to app callback route -> isAuthenticated$ emits true value
4. change route to anything else (protected or not) -> isAuthenticated$ emits false value

**Comment by dbl81** [comment]
I dont know how exactly, but I solved my problem...now I have to investigate why it works now.

Again thank you for your support.

**Comment by Spinnenzunge** [comment]
> The intention of the method was: You call it, you get the value when you want it, once. We also have the other stream which should emit the value like a stream. So you can choose on your end what you want.

The Problem is that the stream only exists for `isAuthenticated$` and `userData$` not for e.g. resolved id token payload. So if we do want to have an observable that acts like a real stream for subscription we would need to create our own one by switchMapping on e.g. isAuthenticated$ or something like that. This is of course possible but would appreciate it if the lib public api would work like that out of the gate.

**Comment by FabianGosebrink** [comment]
We can implement that. However, currently I do not even have time to breathe. We are accepting PRs and when I find time, I will have a look into it :)

**Comment by Michi-2142** [comment]
The get method should be no observable at all, if it is no stream.
